### PR TITLE
Batch process on users

### PR DESCRIPTION
**Pull request by brianteeman**
Pull request for #32732

The batch process on users lets you remove a user from a group. But it doesn't check to make sure that the user is still in at least one group.

### Test 1
1. Create a new user and put them in the registered usergroup
2. Using the batch process try to remove them from the registered usergroup
#### Before
Success message and user not in any group
#### After
Batch process failed with following error: A user must belong to at least one group.

### Test 2
1. Create a new user and put them in the registered AND author usergroup
2. Using the batch process try to remove them from the editor usergroup
#### Before
Success message and user still in the registered and author usergroups
#### After
Batch process failed with following error: The selected user(s) are not in the selected group.

### Test 3
1. Create a new user and put them in the registered AND author usergroup
2. Using the batch process try to remove them from the registered usergroup
#### Before
Success message and user only in the author usergroup
#### After
Success message and user only in the author usergroup

#### Comments
This type of code is really not my strong point - it may need optimizing etc

### CODE REVIEW

File: administrator/components/com_users/src/Model/UserModel.php [hunk 1]
```diff
@@ -716,18 +716,53 @@ public function batchUser($groupId, $userIds, $action)
...
+                $db->setQuery($query);
+                $users = $db->loadColumn();
 
-            // Remove users from the group
-            $query->delete($db->quoteName('#__user_usergroup_map'))
-                ->whereIn($db->quoteName('user_id'), $userIds);
+                // If we have no users to process, throw an error to notify the user
+                if (empty($users)) {
+                    $this->setError(Text::_('COM_USERS_ERROR_NOT_IN_GROUP'));
+                    return false;
```
Comment (sandewt) [inline administrator/components/com_users/src/Model/UserModel.php]: Be consistent, use an empty line

```
 $this->setError(Text::_('COM_USERS_ERROR_NOT_IN_GROUP'));

 return false;
```

Comment (brianteeman) [reply administrator/components/com_users/src/Model/UserModel.php]: Tell the codestyle not me.

**Comment by Franzwohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 2081df5b0433daf70762ff5fdb90480b00dddd8b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38596">issues.joomla.org/tracker/joomla-cms/38596</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 2081df5b0433daf70762ff5fdb90480b00dddd8b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38596">issues.joomla.org/tracker/joomla-cms/38596</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/38596">issues.joomla.org/tracker/joomla-cms/38596</a>.</sub>

File: administrator/components/com_users/src/Model/UserModel.php (hunk 1 above)
Comment (roland-d) [reply administrator/components/com_users/src/Model/UserModel.php]: He has a point, in this PR you use both styles in the same file.

Line 736 has an empty line before the `return` statement and line 751 does not. 

> Tell the codestyle not me.

All I could find was:
```
Blank lines MAY be added to improve readability and to indicate related blocks of code except where explicitly forbidden.
```

I do think we should be consistent, either we add a blank line or we do not. I have no preference as long as  it is consistent.

Comment (brianteeman) [reply administrator/components/com_users/src/Model/UserModel.php]: my point was that the codestyle checker should have picked this up.

**Comment by chmst** [comment]
I remember a rule that comments have to start with a capital lettter?

**Comment by brianteeman** [comment]
@chmst yes you're right my bad. also the multiline comment should not use //

suprised the code style check didnt pick this up.

I will update shortly

**Comment by fancyFranci** [comment]
Thank you! Great branch name :D

**Comment by brianteeman** [comment]
Wow this gets merged but removing unused CSS doesn't. Makes no sense

**Pull request merged**